doc/module-system/module-system.chapter.md

```diff
@@ -0,0 +1,69 @@
+# Module System {#module-system}
+
+## Introduction {#module-system-introduction}
+
+The module system is a language for handling configuration, implemented as a Nix library.
+
+Compared to plain Nix, it adds documentation, type checking and composition or extensibility.
+
+NOTE: This chapter is new and not complete yet. For a gentle introduction to the module system, in the context of NixOS, see [Writing NixOS Modules](https://nixos.org/manual/nixos/unstable/index.html#sec-writing-modules) in the NixOS manual.
+
+## `lib.evalModules` {#module-system-lib-evalModules}
+
+Evaluate a set of modules.  The result is a set with the attributes:
+
+### Parameters {#module-system-lib-evalModules-parameters}
+
+#### `modules` {#module-system-lib-evalModules-param-modules}
+
+A list of modules. These are merged together using various methods to form the final configuration.
+
+#### `specialArgs` {#module-system-lib-evalModules-param-specialArgs}
+
+An attribute set of module arguments that can be used in `imports`.
+
+This is in contrast to `config._module.args`, which is only available within the module fixpoint, which does not exist before all imports are resolved.
+
+#### `specialArgs.class` {#module-system-lib-evalModules-param-specialArgs-class}
+
+If the `class` attribute is set in `specialArgs`, the module system will rejected modules with a different `class`.
+
+This improves the error message that users will encounter when they import an incompatible module that was designed for a different class of configurations.
+
+The `class` value should be in camelcase, and, if applicable, it should match the prefix of the attributes used in (experimental) flakes. Some examples are:
```

Comment:
```suggestion
The `class` value should be in camelcase.

If applicable, it should match the prefix of the attributes used in flakes (experimental). Examples:
```

I'm not sure what you mean by prefix though. Is it the attrset name containing the actual module?

Please link to a sensible definition of camel case. Also link to flakes, otherwise this will be another instance where people will be left wondering what those damn flakes are even supposed to be.

Reply:
I've put scare quotes around prefix to hint that it's a local, informal definition. An example follows and explains it.